### DISCUSSION TITLE
:not() selector missing

### DISCUSSION THREAD

**Discussion by stefan-garofalo** [tailwindlabs/tailwindcss#13767] [Ideas]
Recently in Tailwind 3.4 support for `:has() `selector was implemented. Today styling a card I noticed there isnt support yet for `:not()` as well. 

Since we now have variants like `has-[]:` I figure there should also be `not-[]:`.
Wasnt it just needed or requested enough? Or is there something not so trivial about this implementation?

**Comment by brandonmcconnell** [comment]
@stefan-garofalo:

@MichaelAllenWarner and I collaborated on a PR (#7976 & https://github.com/MichaelAllenWarner/tailwindcss/pull/6) for this a while back, but a lot of it came down to getting the specificity right and making sure that `not` variants always precede non`-not` variants in order of priority. The team opted to hold off on adding this to v3.x at the time.

That said, an official `not` variant is already in the works and will be included in the upcoming Tailwind CSS v4 release:
https://tailwindcss.com/blog/tailwindcss-v4-alpha#composable-variants
> The new architecture makes it possible to compose together variants that act on other selectors, like `group-*`, `peer-*`, `has-*`, and a new `not-*` variant we’re introducing for v4.

**Comment by brandonmcconnell** [comment]
This is now fully supported in Tailwind v4.x. Upgrade today to start using it!
